### ISSUE TITLE
Grade inquiry can leak other student's inquiries.

### ISSUE THREAD

**Issue by ghost** [Submitty/Submitty#6209]
Hi Prof. Cutler!

Bug: The grade inquiry feature can potentially show other student's inquiries and the TA responses.

How to reproduce:

1. Keep the gradable page open when inquiries are open.
2. When other student open inquiries, they will appear on the page as if they were your own (showing the other student's name).
3. TA responses will show up underneath as they answer.

I'm in `CSCI-2500` this semester and went to use the grade inquiry feature for an exam, when I suddenly saw my classmate's grade inquiry for the exam. I have kept the inquiry page open for about 15 minutes and am seeing new inquiries as they roll in. Speaking without any experience with submitty internals, I'm guessing the websocket that sends chat responses sends it to everyone with the page open (instead of the intended student only). This is a breach of privacy for obvious reasons.

Here is a screenshot of the bug: The first box is another student's grade inquiry. Below that is the TA's response to a (possibly) different inquiry.

![gradeinquiry](https://user-images.githubusercontent.com/70179370/108121009-1ff91e00-7070-11eb-83ed-8cb61ba7b7ed.jpg)

Thanks for all your hard work!

**Comment by williamjallen** [comment]
I can confirm that I have seen the same behavior this semester a few times as well as once in Data Structures last semester.

**Comment by Xystem4** [comment]
To add some more hopefully relevant information, I've had this happen to me multiple times in the past where I: opened an exam page once the grade was returned (but grade inquiries were not yet open), and kept the page open until inquiries were open (yes I've kept chrome tabs open for weeks before). Upon refreshing the page after inquiries were opened, I was able to see other students' inquiries (and responses from graders). At this point leaving entering the exam page from another tab would not be showing other students' inquiries.

I'm not too knowledgeable on the actual infrastructure behind any of this or if those steps are actually relevant in reproducing the issue, but the three times I've seen other students' inquiries it's always been after doing that.
